src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -689,6 +689,45 @@ public static void WriteAllBytes(string path, byte[] bytes)
             RandomAccess.WriteAtOffset(sfh, bytes, 0);
         }
 
+        public static void AppendAllBytes(string path, byte[] bytes)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(path);
+            ArgumentNullException.ThrowIfNull(bytes);
+
+            using SafeFileHandle fileHandle = OpenHandle(path, FileMode.Open, FileAccess.Write, FileShare.Read);
+
+            if (fileHandle.CanSeek == false)
+            {
+                throw new NotSupportedException();
+            }
```

Comment:
`RandomAccess.GetLength` already verifies the handle.

Reply:
Oh, ok. I removed the check as a separate commit.